ledger/participant-integration-api/src/main/scala/platform/store/DbSupport.scala

```diff
@@ -20,21 +21,38 @@ case class DbSupport(
 )
 
 object DbSupport {
-  def owner(
+  case class ConnectionPoolConfig(
+      minimumIdle: Int,
+      maxPoolSize: Int,
+      connectionTimeout: FiniteDuration,
+  )
+
+  case class DbConfig(
       jdbcUrl: String,
+      connectionPool: ConnectionPoolConfig,
+      postgres: PostgresDataSourceConfig = PostgresDataSourceConfig(),
```

Comment:
We need to work on postgres and oracle so it looks strange to have a field called `postgres`.

Reply:
Maybe it should be `dataSourceconfig: DataSourceConfig` with variants for Postgres/Oracle.  I don't understand why we would need specific config for one DB but not another (in the general case).

Reply:
I tend to not use polymorphic objects within configuration as it adds complexity on parsing, deriving and understanding of the concrete type used to map for this path - this is why it is within `postgres` path - you know exactly what are the possible values and their types. In case of generic "data-source-config" path and interface used to map for this path - how would parser know which config type to use here? Usually `type` hint is used in this case, but not sure I would like something like:
```
database {
  jdbc-url = "jdbc:h2:mem:default;db_close_delay=-1;db_close_on_exit=false"
  datasource-config {
    type = postgres
    synchronous-commit = off
    tcp-keepalives-count = 5
    tcp-keepalives-idle = 10
    tcp-keepalives-interval = 1
  }
}
```

Let me investigate further if there is a better way for that. I might come with another proposal.